src/PackageVersions.php

```diff
@@ -70,7 +70,10 @@ static function ($filename): array {
             return new self($allPackageLists ? array_shift($allPackageLists) : []);
         }
 
-        return new self(array_merge(...$allPackageLists));
+        /** @var array<string, string> $mergedPackagesList */
```

Comment:
Can this not be inferred at all? :thinking:

Reply:
When creating an instance of [Laminas\Migration\PackageVersions](https://github.com/laminas/laminas-migration/blob/1.6.x/src/PackageVersions.php) the argument passed to the constructor is an array that should already have the key and value as strings.

Another solution would be to remove the `<string, string>` type hint from the constructor docblock, but IMHO that could cause unexpected issues down the road.

That's why I went with the solution to specify the array structure before returning the merged packages list.😃